js/views/pageNavVw.js

```diff
@@ -230,54 +284,94 @@ module.exports = Backbone.View.extend({
     }
   },
 
-  navNotificationsClick: function(e){
-    e.stopPropagation();
-    this.setNotificationCount("");
-    var targ = this.$el.find('.js-navNotificationsMenu');
-    targ.siblings('.popMenu').addClass('hide');
-    if(targ.hasClass('hide')){
-      targ.removeClass('hide').addClass('popMenu-notifications-opened');
-      $('#overlay').removeClass('hide');
-      $('html').on('click.closeNav', function(e){
-        if($(e.target).closest(targ).length === 0){
-          targ.addClass('hide').removeClass('popMenu-notifications-opened');
-          $('#overlay').addClass('hide');
-          $(this).off('.closeNav');
-        }
-      });
-    }else{
-      targ.addClass('hide');
-      $('#overlay').addClass('hide');
-    }
+  closeNotificationsMenu: function() {
+    this.$notifMenu.removeClass('popMenu-opened');
+  },
+
+  isNotifMenuOpen: function() {
+    return this.$notifMenu.hasClass('popMenu-opened');
   },
 
   setNotificationCount: function(count){
-    if(count > 99) {
-      count = "..";
+    if (isNaN(parseInt(count))) return;
+
+    if (count > 99) {
+      count = '..';
     }
-    this.$el.find('.js-navNotifications .badge').attr('data-count', count);
-  },
-
-  navProfileClick: function(e){
-    e.stopPropagation();
-    var targ = this.$el.find('.js-navProfileMenu');
-    targ.siblings('.popMenu').addClass('hide');
-    if(targ.hasClass('hide')){
-      targ.removeClass('hide').addClass('popMenu-navBar-opened');
-      $('#overlay').removeClass('hide');
-      $('html').on('click.closeNav', function(e){
-        if($(e.target).closest(targ).length === 0){
-          targ.addClass('hide').removeClass('popMenu-navBar-opened');
-          $('#overlay').addClass('hide');
-          $(this).off('.closeNav');
-        }
+
+    if (count == 0) {
+      this.$navNotif.find('.badge')
+          .removeAttr('data-count', count);
+    } else {
+      this.$navNotif.find('.badge')
+          .attr('data-count', count);
+    }
+  },
+
+  onNotifMenuClose: function() {
+    this.notificationsVw.resetScroll();
+  },
+
+  onNotifMenuOpen: function() {
```

Comment:
This means you never see an unread notification in the list when you open it. What should happen instead is the notifications aren't marked as read until you close the notifications list (that's how it worked previously).